feat: Add `checkJSDoc` option to multiline-comment-style (#16807)
eslint/eslint: docs/src/rules/multiline-comment-style.md

* `"starred-block"` (default): Disallows consecutive line comments in favor of block comments. Additionally, requires block comments to have an aligned `*` character before each line.
* `"bare-block"`: Disallows consecutive line comments in favor of block comments, and disallows block comments from having a `"*"` character before each line.

> * `"separate-lines"`: Disallows block comments in favor of consecutive line comments. By default, this rule ignores JSDoc comments. To disallow them too, set the `allowJSDoc` option to `false`.

Reviewer: Wording here is a little confusing, maybe this helps?
```suggestion
* `"separate-lines"`: Disallows block comments in favor of consecutive line comments. By default, this option ignores JSDoc comments. To also apply this rule to JSDoc comments, set the `allowJSDoc` option to `false`.
```
